{
  "nbformat": 4,
  "nbformat_minor": 0,
  "metadata": {
    "colab": {
      "provenance": [],
      "gpuType": "V5E1"
    },
    "kernelspec": {
      "name": "python3",
      "display_name": "Python 3"
    },
    "language_info": {
      "name": "python"
    },
    "accelerator": "TPU"
  },
  "cells": [
    {
      "cell_type": "code",
      "execution_count": 4,
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DrLIVRwjjyPU",
        "outputId": "8bcf4db4-29d1-4598-8525-ba516ac7636b"
      },
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\r0% [Working]\r            \rHit:1 https://cli.github.com/packages stable InRelease\n",
            "\r0% [Connecting to archive.ubuntu.com (185.125.190.83)] [Waiting for headers] [W\r                                                                               \rGet:2 https://cloud.r-project.org/bin/linux/ubuntu jammy-cran40/ InRelease [3,632 B]\n",
            "Get:3 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  InRelease [1,581 B]\n",
            "Get:4 http://security.ubuntu.com/ubuntu jammy-security InRelease [129 kB]\n",
            "Get:5 https://r2u.stat.illinois.edu/ubuntu jammy InRelease [6,555 B]\n",
            "Hit:6 http://archive.ubuntu.com/ubuntu jammy InRelease\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy-updates InRelease [128 kB]\n",
            "Get:8 https://developer.download.nvidia.com/compute/cuda/repos/ubuntu2204/x86_64  Packages [2,125 kB]\n",
            "Get:9 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy InRelease [18.1 kB]\n",
            "Get:10 https://r2u.stat.illinois.edu/ubuntu jammy/main amd64 Packages [2,826 kB]\n",
            "Hit:11 https://ppa.launchpadcontent.net/graphics-drivers/ppa/ubuntu jammy InRelease\n",
            "Get:12 http://archive.ubuntu.com/ubuntu jammy-backports InRelease [127 kB]\n",
            "Hit:13 https://ppa.launchpadcontent.net/ubuntugis/ppa/ubuntu jammy InRelease\n",
            "Get:14 https://r2u.stat.illinois.edu/ubuntu jammy/main all Packages [9,437 kB]\n",
            "Get:15 http://security.ubuntu.com/ubuntu jammy-security/main amd64 Packages [3,526 kB]\n",
            "Get:16 https://ppa.launchpadcontent.net/deadsnakes/ppa/ubuntu jammy/main amd64 Packages [38.5 kB]\n",
            "Get:17 http://archive.ubuntu.com/ubuntu jammy-updates/main amd64 Packages [3,856 kB]\n",
            "Fetched 22.2 MB in 5s (4,927 kB/s)\n",
            "Reading package lists... Done\n",
            "W: Skipping acquire of configured file 'main/source/Sources' as repository 'https://r2u.stat.illinois.edu/ubuntu jammy InRelease' does not seem to provide it (sources.list entry misspelt?)\n",
            "Reading package lists... Done\n",
            "Building dependency tree... Done\n",
            "Reading state information... Done\n",
            "build-essential is already the newest version (12.9ubuntu3).\n",
            "unzip is already the newest version (6.0-26ubuntu3.2).\n",
            "The following additional packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "Suggested packages:\n",
            "  blends-doc menu-l10n gksu | kde-runtime | ktsuss\n",
            "The following NEW packages will be installed:\n",
            "  blends-common libkdf5-2 libncbi-vdb2 libncbi-wvdb2 med-config menu\n",
            "  sra-toolkit\n",
            "0 upgraded, 7 newly installed, 0 to remove and 43 not upgraded.\n",
            "Need to get 8,290 kB of archives.\n",
            "After this operation, 23.0 MB of additional disk space will be used.\n",
            "Get:1 http://archive.ubuntu.com/ubuntu jammy/universe amd64 menu amd64 2.1.47ubuntu4 [354 kB]\n",
            "Get:2 http://archive.ubuntu.com/ubuntu jammy/universe amd64 blends-common all 0.7.4ubuntu1 [15.9 kB]\n",
            "Get:3 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libkdf5-2 amd64 2.11.2+dfsg-4build2 [14.7 kB]\n",
            "Get:4 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-vdb2 amd64 2.11.2+dfsg-4build2 [1,364 kB]\n",
            "Get:5 http://archive.ubuntu.com/ubuntu jammy/universe amd64 libncbi-wvdb2 amd64 2.11.2+dfsg-4build2 [1,252 kB]\n",
            "Get:6 http://archive.ubuntu.com/ubuntu jammy/universe amd64 sra-toolkit amd64 2.11.3+dfsg-1ubuntu1 [5,277 kB]\n",
            "Get:7 http://archive.ubuntu.com/ubuntu jammy/universe amd64 med-config all 3.7.1 [11.6 kB]\n",
            "Fetched 8,290 kB in 2s (5,410 kB/s)\n",
            "Preconfiguring packages ...\n",
            "Selecting previously unselected package menu.\n",
            "(Reading database ... 121229 files and directories currently installed.)\n",
            "Preparing to unpack .../0-menu_2.1.47ubuntu4_amd64.deb ...\n",
            "Unpacking menu (2.1.47ubuntu4) ...\n",
            "Selecting previously unselected package blends-common.\n",
            "Preparing to unpack .../1-blends-common_0.7.4ubuntu1_all.deb ...\n",
            "Unpacking blends-common (0.7.4ubuntu1) ...\n",
            "Selecting previously unselected package libkdf5-2.\n",
            "Preparing to unpack .../2-libkdf5-2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-vdb2.\n",
            "Preparing to unpack .../3-libncbi-vdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package libncbi-wvdb2.\n",
            "Preparing to unpack .../4-libncbi-wvdb2_2.11.2+dfsg-4build2_amd64.deb ...\n",
            "Unpacking libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Selecting previously unselected package sra-toolkit.\n",
            "Preparing to unpack .../5-sra-toolkit_2.11.3+dfsg-1ubuntu1_amd64.deb ...\n",
            "Unpacking sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Selecting previously unselected package med-config.\n",
            "Preparing to unpack .../6-med-config_3.7.1_all.deb ...\n",
            "Unpacking med-config (3.7.1) ...\n",
            "Setting up libncbi-vdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libkdf5-2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up libncbi-wvdb2 (2.11.2+dfsg-4build2) ...\n",
            "Setting up menu (2.1.47ubuntu4) ...\n",
            "Setting up blends-common (0.7.4ubuntu1) ...\n",
            "Setting up sra-toolkit (2.11.3+dfsg-1ubuntu1) ...\n",
            "Setting up med-config (3.7.1) ...\n",
            "Adding group `med' (GID 107) ...\n",
            "Done.\n",
            "Processing triggers for man-db (2.10.2-1) ...\n",
            "Processing triggers for libc-bin (2.35-0ubuntu3.8) ...\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm_debug.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libumf.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_opencl.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_loader.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libhwloc.so.15 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtcm.so.1 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbmalloc_proxy.so.2 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_0.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbbbind_2_5.so.3 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libur_adapter_level_zero_v2.so.0 is not a symbolic link\n",
            "\n",
            "/sbin/ldconfig.real: /usr/local/lib/libtbb.so.12 is not a symbolic link\n",
            "\n",
            "Processing triggers for menu (2.1.47ubuntu4) ...\n",
            "--2025-11-11 07:05:11--  https://github.com/alexdobin/STAR/archive/2.7.10a.zip\n",
            "Resolving github.com (github.com)... 140.82.113.3\n",
            "Connecting to github.com (github.com)|140.82.113.3|:443... connected.\n",
            "HTTP request sent, awaiting response... 302 Found\n",
            "Location: https://codeload.github.com/alexdobin/STAR/zip/refs/tags/2.7.10a [following]\n",
            "--2025-11-11 07:05:11--  https://codeload.github.com/alexdobin/STAR/zip/refs/tags/2.7.10a\n",
            "Resolving codeload.github.com (codeload.github.com)... 140.82.113.9\n",
            "Connecting to codeload.github.com (codeload.github.com)|140.82.113.9|:443... connected.\n",
            "HTTP request sent, awaiting response... 200 OK\n",
            "Length: unspecified [application/zip]\n",
            "Saving to: ‘2.7.10a.zip’\n",
            "\n",
            "2.7.10a.zip             [       <=>          ]  11.86M  6.84MB/s    in 1.7s    \n",
            "\n",
            "2025-11-11 07:05:13 (6.84 MB/s) - ‘2.7.10a.zip’ saved [12437043]\n",
            "\n",
            "Archive:  2.7.10a.zip\n",
            "7f0f3f16ebf232963e99891b5d3fecba071bdd57\n",
            "   creating: STAR-2.7.10a/\n",
            "  inflating: STAR-2.7.10a/.gitignore  \n",
            " extracting: STAR-2.7.10a/.gitmodules  \n",
            "  inflating: STAR-2.7.10a/.travis.yml  \n",
            "  inflating: STAR-2.7.10a/CHANGES.md  \n",
            "  inflating: STAR-2.7.10a/CODE_OF_CONDUCT.md  \n",
            "  inflating: STAR-2.7.10a/CONTRIBUTING.md  \n",
            "  inflating: STAR-2.7.10a/LICENSE    \n",
            "  inflating: STAR-2.7.10a/README.md  \n",
            "  inflating: STAR-2.7.10a/RELEASEnotes.md  \n",
            "  inflating: STAR-2.7.10a/_config.yml  \n",
            "   creating: STAR-2.7.10a/bin/\n",
            "   creating: STAR-2.7.10a/bin/Linux_x86_64/\n",
            "  inflating: STAR-2.7.10a/bin/Linux_x86_64/STAR  \n",
            "  inflating: STAR-2.7.10a/bin/Linux_x86_64/STARlong  \n",
            "   creating: STAR-2.7.10a/bin/Linux_x86_64_static/\n",
            "  inflating: STAR-2.7.10a/bin/Linux_x86_64_static/STAR  \n",
            "  inflating: STAR-2.7.10a/bin/Linux_x86_64_static/STARlong  \n",
            "   creating: STAR-2.7.10a/bin/MacOSX_x86_64/\n",
            "  inflating: STAR-2.7.10a/bin/MacOSX_x86_64/STAR  \n",
            "  inflating: STAR-2.7.10a/bin/MacOSX_x86_64/STARlong  \n",
            "   creating: STAR-2.7.10a/doc/\n",
            "  inflating: STAR-2.7.10a/doc/STARmanual.pdf  \n",
            "   creating: STAR-2.7.10a/docs/\n",
            "  inflating: STAR-2.7.10a/docs/STARconsensus.md  \n",
            "  inflating: STAR-2.7.10a/docs/STARsolo.md  \n",
            "   creating: STAR-2.7.10a/extras/\n",
            "   creating: STAR-2.7.10a/extras/doc-latex/\n",
            "  inflating: STAR-2.7.10a/extras/doc-latex/STARmanual.tex  \n",
            "  inflating: STAR-2.7.10a/extras/doc-latex/convertParDefToLatexTable.awk  \n",
            "  inflating: STAR-2.7.10a/extras/doc-latex/parametersDefault.tex  \n",
            "   creating: STAR-2.7.10a/extras/docker/\n",
            "  inflating: STAR-2.7.10a/extras/docker/Dockerfile  \n",
            "   creating: STAR-2.7.10a/extras/parameters/\n",
            "  inflating: STAR-2.7.10a/extras/parameters/ENCODE.txt  \n",
            "   creating: STAR-2.7.10a/extras/scripts/\n",
            "  inflating: STAR-2.7.10a/extras/scripts/calcInsertCoverage.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/extractSJfromGTF.sh  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/filterCirc.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/mergeGeneCounts.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/mergeLogFinal.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/mergeSuperContig.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjBED12.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjCollapseSamples.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjFromSAMcollapseUandM.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjFromSAMcollapseUandM_inclOverlaps.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/sjMotif.m  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloBasicCellFilter.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloCompareMtx.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloCountMatrixFromBAM.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloExtractFiltCells.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/soloUMIperCell.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/tagXSstrandedData.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/transcriptTypes.awk  \n",
            "  inflating: STAR-2.7.10a/extras/scripts/transfragsFromBedGraph.awk  \n",
            "   creating: STAR-2.7.10a/extras/tests/\n",
            "   creating: STAR-2.7.10a/extras/tests/scripts/\n",
            "  inflating: STAR-2.7.10a/extras/tests/scripts/checkCellReadsStats.awk  \n",
            "  inflating: STAR-2.7.10a/extras/tests/scripts/checkCellReadsStats_vsBAM.awk  \n",
            "  inflating: STAR-2.7.10a/extras/tests/scripts/checkCellReadsStats_vsMatrix.awk  \n",
            "   creating: STAR-2.7.10a/source/\n",
            "  inflating: STAR-2.7.10a/source/AlignVsTranscript.h  \n",
            "  inflating: STAR-2.7.10a/source/BAMbinSortByCoordinate.cpp  \n",
            "  inflating: STAR-2.7.10a/source/BAMbinSortByCoordinate.h  \n",
            "  inflating: STAR-2.7.10a/source/BAMbinSortUnmapped.cpp  \n",
            "  inflating: STAR-2.7.10a/source/BAMbinSortUnmapped.h  \n",
            "  inflating: STAR-2.7.10a/source/BAMfunctions.cpp  \n",
            "  inflating: STAR-2.7.10a/source/BAMfunctions.h  \n",
            "  inflating: STAR-2.7.10a/source/BAMoutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/BAMoutput.h  \n",
            "  inflating: STAR-2.7.10a/source/Chain.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Chain.h  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign.h  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign_chimericBAMoutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign_chimericJunctionOutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericAlign_chimericStitching.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericDetection.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericDetection.h  \n",
            "  inflating: STAR-2.7.10a/source/ChimericDetection_chimericDetectionMult.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericSegment.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ChimericSegment.h  \n",
            "  inflating: STAR-2.7.10a/source/ChimericTranscript.h  \n",
            "  inflating: STAR-2.7.10a/source/ClipCR4.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ClipCR4.h  \n",
            "  inflating: STAR-2.7.10a/source/ClipMate.h  \n",
            "  inflating: STAR-2.7.10a/source/ClipMate_clip.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ClipMate_clipChunk.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ClipMate_initialize.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ErrorWarning.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ErrorWarning.h  \n",
            "  inflating: STAR-2.7.10a/source/GTF.cpp  \n",
            "  inflating: STAR-2.7.10a/source/GTF.h  \n",
            "  inflating: STAR-2.7.10a/source/GTF_superTranscript.cpp  \n",
            "  inflating: STAR-2.7.10a/source/GTF_transcriptGeneSJ.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome.h  \n",
            "  inflating: STAR-2.7.10a/source/Genome_genomeGenerate.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome_genomeLoad.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome_genomeOutLoad.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome_insertSequences.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Genome_transformGenome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/GlobalVariables.cpp  \n",
            "  inflating: STAR-2.7.10a/source/GlobalVariables.h  \n",
            "  inflating: STAR-2.7.10a/source/InOutStreams.cpp  \n",
            "  inflating: STAR-2.7.10a/source/InOutStreams.h  \n",
            "  inflating: STAR-2.7.10a/source/IncludeDefine.h  \n",
            "  inflating: STAR-2.7.10a/source/Makefile  \n",
            "  inflating: STAR-2.7.10a/source/OutSJ.cpp  \n",
            "  inflating: STAR-2.7.10a/source/OutSJ.h  \n",
            "  inflating: STAR-2.7.10a/source/PackedArray.cpp  \n",
            "  inflating: STAR-2.7.10a/source/PackedArray.h  \n",
            "  inflating: STAR-2.7.10a/source/ParameterInfo.h  \n",
            "  inflating: STAR-2.7.10a/source/Parameters.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters.h  \n",
            "  inflating: STAR-2.7.10a/source/ParametersChimeric.h  \n",
            "  inflating: STAR-2.7.10a/source/ParametersChimeric_initialize.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ParametersClip.h  \n",
            "  inflating: STAR-2.7.10a/source/ParametersClip_initialize.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ParametersGenome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ParametersGenome.h  \n",
            "  inflating: STAR-2.7.10a/source/ParametersSolo.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ParametersSolo.h  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_closeReadsFiles.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_openReadsFiles.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_readFilesInit.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_readSAMheader.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Parameters_samAttributes.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Quantifications.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Quantifications.h  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign.h  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlignChunk.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlignChunk.h  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlignChunk_mapChunk.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlignChunk_processChunks.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_CIGAR.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_alignBAM.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_assignAlignToWindow.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_calcCIGAR.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_chimericDetection.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_chimericDetectionOld.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_chimericDetectionOldOutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_chimericDetectionPEmerged.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_createExtendWindowsWithAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_mapOneRead.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_mapOneReadSpliceGraph.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_mappedFilter.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_maxMappableLength2strands.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_multMapSelect.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_oneRead.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputAlignments.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputSpliceGraphSAM.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputTranscriptCIGARp.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputTranscriptSAM.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputTranscriptSJ.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_outputVariation.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_peOverlapMergeMap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_quantTranscriptome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_stitchPieces.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_stitchWindowSeeds.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_storeAligns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_transformGenome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAlign_waspMap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ReadAnnotations.h  \n",
            "  inflating: STAR-2.7.10a/source/STAR.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SequenceFuns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SequenceFuns.h  \n",
            "  inflating: STAR-2.7.10a/source/SharedMemory.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SharedMemory.h  \n",
            "   creating: STAR-2.7.10a/source/SimpleGoodTuring/\n",
            "  inflating: STAR-2.7.10a/source/SimpleGoodTuring/ReadMe.txt  \n",
            "  inflating: STAR-2.7.10a/source/SimpleGoodTuring/sgt.h  \n",
            "  inflating: STAR-2.7.10a/source/SjdbClass.h  \n",
            "  inflating: STAR-2.7.10a/source/Solo.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Solo.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloBarcode.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloBarcode.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloBarcode_extractBarcode.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloCommon.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeatureTypes.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_addBAMtags.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_cellFiltering.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_collapseUMI_Graph.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_collapseUMIall_multiMappers.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_countCBgeneUMI.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_countSmartSeq.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_countVelocyto.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_emptyDrops_CR.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_loadRawMatrix.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_outputResults.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_processRecords.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_quantTranscript.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_redistributeReadsByCB.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_statsOutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFeature_sumThreads.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloFilteredCells.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloRead.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloRead.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadBarcode.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadBarcode.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadBarcodeStats.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadBarcode_getCBandUMI.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeature.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeature.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeatureStats.h  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeature_inputRecords.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloReadFeature_record.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SoloRead_record.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph.h  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph_findSuperTr.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph_swScoreSpliced.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SpliceGraph_swTraceBack.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Stats.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Stats.h  \n",
            "  inflating: STAR-2.7.10a/source/SuffixArrayFuns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SuffixArrayFuns.h  \n",
            "  inflating: STAR-2.7.10a/source/SuperTranscriptome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/SuperTranscriptome.h  \n",
            "  inflating: STAR-2.7.10a/source/Test.hpp  \n",
            "  inflating: STAR-2.7.10a/source/ThreadControl.cpp  \n",
            "  inflating: STAR-2.7.10a/source/ThreadControl.h  \n",
            "  inflating: STAR-2.7.10a/source/TimeFunctions.cpp  \n",
            "  inflating: STAR-2.7.10a/source/TimeFunctions.h  \n",
            "  inflating: STAR-2.7.10a/source/Transcript.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript.h  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_alignScore.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_convertGenomeCigar.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_generateCigarP.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_transformGenome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_variationAdjust.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcript_variationOutput.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome.h  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_alignExonOverlap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_classifyAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_geneCountsAddAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_geneFullAlignOverlap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Transcriptome_quantAlign.cpp  \n",
            " extracting: STAR-2.7.10a/source/VERSION  \n",
            "  inflating: STAR-2.7.10a/source/Variation.cpp  \n",
            "  inflating: STAR-2.7.10a/source/Variation.h  \n",
            "  inflating: STAR-2.7.10a/source/bamRemoveDuplicates.cpp  \n",
            "  inflating: STAR-2.7.10a/source/bamRemoveDuplicates.h  \n",
            "  inflating: STAR-2.7.10a/source/bamSortByCoordinate.cpp  \n",
            "  inflating: STAR-2.7.10a/source/bamSortByCoordinate.h  \n",
            "  inflating: STAR-2.7.10a/source/bam_cat.c  \n",
            "  inflating: STAR-2.7.10a/source/bam_cat.h  \n",
            "  inflating: STAR-2.7.10a/source/binarySearch2.cpp  \n",
            "  inflating: STAR-2.7.10a/source/binarySearch2.h  \n",
            "  inflating: STAR-2.7.10a/source/blocksOverlap.cpp  \n",
            "  inflating: STAR-2.7.10a/source/blocksOverlap.h  \n",
            "  inflating: STAR-2.7.10a/source/extendAlign.cpp  \n",
            "  inflating: STAR-2.7.10a/source/extendAlign.h  \n",
            "  inflating: STAR-2.7.10a/source/funCompareUintAndSuffixes.cpp  \n",
            "  inflating: STAR-2.7.10a/source/funCompareUintAndSuffixes.h  \n",
            "  inflating: STAR-2.7.10a/source/funCompareUintAndSuffixesMemcmp.cpp  \n",
            "  inflating: STAR-2.7.10a/source/funCompareUintAndSuffixesMemcmp.h  \n",
            "  inflating: STAR-2.7.10a/source/genomeGenerate.h  \n",
            "  inflating: STAR-2.7.10a/source/genomeParametersWrite.cpp  \n",
            "  inflating: STAR-2.7.10a/source/genomeParametersWrite.h  \n",
            "  inflating: STAR-2.7.10a/source/genomeSAindex.cpp  \n",
            "  inflating: STAR-2.7.10a/source/genomeSAindex.h  \n",
            "  inflating: STAR-2.7.10a/source/genomeScanFastaFiles.cpp  \n",
            "  inflating: STAR-2.7.10a/source/genomeScanFastaFiles.h  \n",
            "   creating: STAR-2.7.10a/source/htslib/\n",
            "  inflating: STAR-2.7.10a/source/htslib/.gitignore  \n",
            "  inflating: STAR-2.7.10a/source/htslib/.travis.yml  \n",
            "  inflating: STAR-2.7.10a/source/htslib/Makefile  \n",
            "  inflating: STAR-2.7.10a/source/htslib/README.md  \n",
            "  inflating: STAR-2.7.10a/source/htslib/bgzf.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/bgzip.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/config.h  \n",
            "   creating: STAR-2.7.10a/source/htslib/cram/\n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_codecs.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_codecs.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_decode.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_decode.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_encode.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_encode.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_index.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_index.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_io.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_io.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_samtools.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_samtools.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_stats.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_stats.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/cram_structs.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/files.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/mFILE.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/mFILE.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/md5.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/md5.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/misc.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/open_trace_file.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/open_trace_file.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/os.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/pooled_alloc.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/pooled_alloc.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/sam_header.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/sam_header.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/string_alloc.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/string_alloc.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/thread_pool.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/thread_pool.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/vlen.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/vlen.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/zfio.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/cram/zfio.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/faidx.5  \n",
            "  inflating: STAR-2.7.10a/source/htslib/faidx.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/hfile.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/hfile_internal.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/hfile_net.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/hts.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib.mk  \n",
            "   creating: STAR-2.7.10a/source/htslib/htslib/\n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/bgzf.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/faidx.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/hfile.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/hts.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/hts_defs.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/kfunc.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/khash.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/khash_str2int.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/klist.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/knetfile.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/kseq.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/ksort.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/kstdint.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/kstring.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/sam.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/synced_bcf_reader.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/tbx.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/vcf.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/vcf_sweep.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib/vcfutils.h  \n",
            "  inflating: STAR-2.7.10a/source/htslib/htslib_vars.mk  \n",
            "  inflating: STAR-2.7.10a/source/htslib/kfunc.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/knetfile.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/kstring.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/sam.5  \n",
            "  inflating: STAR-2.7.10a/source/htslib/sam.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/synced_bcf_reader.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/tabix.1  \n",
            "  inflating: STAR-2.7.10a/source/htslib/tabix.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/tbx.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/vcf.5  \n",
            "  inflating: STAR-2.7.10a/source/htslib/vcf.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/vcf_sweep.c  \n",
            "  inflating: STAR-2.7.10a/source/htslib/vcfutils.c  \n",
            "  inflating: STAR-2.7.10a/source/insertSeqSA.cpp  \n",
            "  inflating: STAR-2.7.10a/source/insertSeqSA.h  \n",
            "  inflating: STAR-2.7.10a/source/mapThreadsSpawn.cpp  \n",
            "  inflating: STAR-2.7.10a/source/mapThreadsSpawn.h  \n",
            "   creating: STAR-2.7.10a/source/opal/\n",
            "  inflating: STAR-2.7.10a/source/opal/LICENSE  \n",
            "  inflating: STAR-2.7.10a/source/opal/opal.cpp  \n",
            "  inflating: STAR-2.7.10a/source/opal/opal.h  \n",
            "  inflating: STAR-2.7.10a/source/opal/simde_avx2.h  \n",
            "  inflating: STAR-2.7.10a/source/outputSJ.cpp  \n",
            "  inflating: STAR-2.7.10a/source/outputSJ.h  \n",
            "  inflating: STAR-2.7.10a/source/parametersDefault  \n",
            "  inflating: STAR-2.7.10a/source/parametersDefault.xxd  \n",
            "  inflating: STAR-2.7.10a/source/readBarcodeLoad.h  \n",
            "  inflating: STAR-2.7.10a/source/readLoad.cpp  \n",
            "  inflating: STAR-2.7.10a/source/readLoad.h  \n",
            "  inflating: STAR-2.7.10a/source/samHeaders.cpp  \n",
            "  inflating: STAR-2.7.10a/source/samHeaders.h  \n",
            "  inflating: STAR-2.7.10a/source/serviceFuns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/signalFromBAM.cpp  \n",
            "  inflating: STAR-2.7.10a/source/signalFromBAM.h  \n",
            "  inflating: STAR-2.7.10a/source/sjAlignSplit.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjAlignSplit.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbBuildIndex.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbBuildIndex.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbInsertJunctions.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbInsertJunctions.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbLoadFromFiles.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbLoadFromFiles.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbLoadFromStream.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbLoadFromStream.h  \n",
            "  inflating: STAR-2.7.10a/source/sjdbPrepare.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sjdbPrepare.h  \n",
            "  inflating: STAR-2.7.10a/source/soloInputFeatureUMI.cpp  \n",
            "  inflating: STAR-2.7.10a/source/soloInputFeatureUMI.h  \n",
            "  inflating: STAR-2.7.10a/source/sortSuffixesBucket.h  \n",
            "  inflating: STAR-2.7.10a/source/stitchAlignToTranscript.cpp  \n",
            "  inflating: STAR-2.7.10a/source/stitchAlignToTranscript.h  \n",
            "  inflating: STAR-2.7.10a/source/stitchGapIndel.cpp  \n",
            "  inflating: STAR-2.7.10a/source/stitchWindowAligns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/stitchWindowAligns.h  \n",
            "  inflating: STAR-2.7.10a/source/streamFuns.cpp  \n",
            "  inflating: STAR-2.7.10a/source/streamFuns.h  \n",
            "  inflating: STAR-2.7.10a/source/stringSubstituteAll.cpp  \n",
            "  inflating: STAR-2.7.10a/source/stringSubstituteAll.h  \n",
            "  inflating: STAR-2.7.10a/source/sysRemoveDir.cpp  \n",
            "  inflating: STAR-2.7.10a/source/sysRemoveDir.h  \n",
            "  inflating: STAR-2.7.10a/source/twoPassRunPass1.cpp  \n",
            "  inflating: STAR-2.7.10a/source/twoPassRunPass1.h  \n",
            "bin\t\t    _config.yml      docs     README.md\n",
            "CHANGES.md\t    CONTRIBUTING.md  extras   RELEASEnotes.md\n",
            "CODE_OF_CONDUCT.md  doc\t\t     LICENSE  source\n",
            "make -C htslib lib-static\n",
            "make[1]: Entering directory '/content/STAR-2.7.10a/source/htslib'\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o kfunc.o kfunc.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o knetfile.o knetfile.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o kstring.o kstring.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o bgzf.o bgzf.c\n",
            "\u001b[01m\u001b[Kbgzf.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kbgzf_write\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kbgzf.c:733:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  733 |     \u001b[01;35m\u001b[Kif\u001b[m\u001b[K ( !fp->is_compressed )\n",
            "      |     \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kbgzf.c:736:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n",
            "  736 |         \u001b[01;36m\u001b[Kconst\u001b[m\u001b[K uint8_t *input = (const uint8_t*)data;\n",
            "      |         \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o faidx.o faidx.c\n",
            "\u001b[01m\u001b[Kfaidx.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kfai_load\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kfaidx.c:265:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kelse\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  265 |     \u001b[01;35m\u001b[Kelse\u001b[m\u001b[K\n",
            "      |     \u001b[01;35m\u001b[K^~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kfaidx.c:268:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kelse\u001b[m\u001b[K’\n",
            "  268 |         \u001b[01;36m\u001b[Kif\u001b[m\u001b[K (fp == 0) {\n",
            "      |         \u001b[01;36m\u001b[K^~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hfile.o hfile.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hfile_net.o hfile_net.c\n",
            "echo '#define HTS_VERSION \"0.0.1\"' > version.h\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o hts.o hts.c\n",
            "\u001b[01m\u001b[Khts.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Khts_idx_init\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Khts.c:522:63:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Koverflow in conversion from ‘\u001b[01m\u001b[Kuint32_t\u001b[m\u001b[K’ {aka ‘\u001b[01m\u001b[Kunsigned int\u001b[m\u001b[K’} to ‘\u001b[01m\u001b[Kint\u001b[m\u001b[K’ changes value from ‘\u001b[01m\u001b[Kidx->z.last_bin = 4294967295\u001b[m\u001b[K’ to ‘\u001b[01m\u001b[K-1\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Woverflow\u0007-Woverflow\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  522 |         idx->z.save_bin = idx->z.save_tid = idx->z.last_tid = \u001b[01;35m\u001b[Kidx\u001b[m\u001b[K->z.last_bin = 0xffffffffu;\n",
            "      |                                                               \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o sam.o sam.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Ksam.c:3\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kbam_hdr_write\u001b[m\u001b[K’ at \u001b[01m\u001b[Ksam.c:130:2\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying 4 bytes from a string of the same length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o synced_bcf_reader.o synced_bcf_reader.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcf_sweep.o vcf_sweep.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o tbx.o tbx.c\n",
            "\u001b[01m\u001b[Ktbx.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ktbx_seqnames\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Ktbx.c:285:5:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kfor\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  285 |     \u001b[01;35m\u001b[Kfor\u001b[m\u001b[K (tid=0; tid<m; tid++)\n",
            "      |     \u001b[01;35m\u001b[K^~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Ktbx.c:287:9:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kfor\u001b[m\u001b[K’\n",
            "  287 |         \u001b[01;36m\u001b[K*\u001b[m\u001b[Kn = m;\n",
            "      |         \u001b[01;36m\u001b[K^\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcf.o vcf.c\n",
            "\u001b[01m\u001b[Kvcf.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kvcf_format\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kvcf.c:1909:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1909 |                         \u001b[01;35m\u001b[Kif\u001b[m\u001b[K ( !first ) kputc(';', s); first = 0;\n",
            "      |                         \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kvcf.c:1909:54:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n",
            " 1909 |                         if ( !first ) kputc(';', s); \u001b[01;36m\u001b[Kfirst\u001b[m\u001b[K = 0;\n",
            "      |                                                      \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kvcf.c:1948:21:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kthis ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’ clause does not guard... [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmisleading-indentation\u0007-Wmisleading-indentation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1948 |                     \u001b[01;35m\u001b[Kif\u001b[m\u001b[K (!first) kputc(':', s); first = 0;\n",
            "      |                     \u001b[01;35m\u001b[K^~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kvcf.c:1948:48:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K...this statement, but the latter is misleadingly indented as if it were guarded by the ‘\u001b[01m\u001b[Kif\u001b[m\u001b[K’\n",
            " 1948 |                     if (!first) kputc(':', s); \u001b[01;36m\u001b[Kfirst\u001b[m\u001b[K = 0;\n",
            "      |                                                \u001b[01;36m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o vcfutils.o vcfutils.c\n",
            "In file included from \u001b[01m\u001b[Khtslib/vcf.h:12\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Khtslib/vcfutils.h:8\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kvcfutils.c:1\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Kvcfutils.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kbcf_remove_alleles\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kvcfutils.c:291:25:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  291 |                 assert( \u001b[01;35m\u001b[Kn\u001b[m\u001b[K=nG_ori );\n",
            "      |                         \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_codecs.o cram/cram_codecs.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_decode.o cram/cram_decode.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_encode.o cram/cram_encode.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_index.o cram/cram_index.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_io.o cram/cram_io.c\n",
            "\u001b[01m\u001b[Kcram/cram_io.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kcram_populate_ref\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/cram_io.c:1529:34:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K.tmp_\u001b[m\u001b[K’ directive writing 5 bytes into a region of size between 1 and 4096 [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wformat-overflow=\u0007-Wformat-overflow=\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            " 1529 |             sprintf(path_tmp, \"%s\u001b[01;35m\u001b[K.tmp_\u001b[m\u001b[K%d\", path, /*getpid(),*/ i);\n",
            "      |                                  \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/cram_io.c:1529:31:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Kdirective argument in the range [0, 2147483647]\n",
            " 1529 |             sprintf(path_tmp, \u001b[01;36m\u001b[K\"%s.tmp_%d\"\u001b[m\u001b[K, path, /*getpid(),*/ i);\n",
            "      |                               \u001b[01;36m\u001b[K^~~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/stdio.h:894\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/cram_io.c:51\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/stdio2.h:38:10:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin___sprintf_chk\u001b[m\u001b[K’ output between 7 and 4111 bytes into a destination of size 4096\n",
            "   38 |   return \u001b[01;36m\u001b[K__builtin___sprintf_chk (__s, __USE_FORTIFY_LEVEL - 1,\u001b[m\u001b[K\n",
            "      |          \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   39 | \u001b[01;36m\u001b[K                                  __glibc_objsize (__s), __fmt,\u001b[m\u001b[K\n",
            "      |                                   \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   40 | \u001b[01;36m\u001b[K                                  __va_arg_pack ())\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;36m\u001b[K~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/cram_io.c:55\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2917:6\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2987:3\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kfull_path\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2909:2\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_write_SAM_hdr\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:2987:3\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 4096 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kcram_dopen\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/cram_io.c:3291:2\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ specified bound 20 equals destination size [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_samtools.o cram/cram_samtools.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/cram_stats.o cram/cram_stats.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/files.o cram/files.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/mFILE.o cram/mFILE.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/md5.o cram/md5.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/open_trace_file.o cram/open_trace_file.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/pooled_alloc.o cram/pooled_alloc.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/sam_header.o cram/sam_header.c\n",
            "In file included from \u001b[01m\u001b[Kcram/sam_header.c:36\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[Kcram/sam_header.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ksam_hdr_add_lines\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/sam_header.c:328:20:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  328 |             assert(\u001b[01;35m\u001b[Kp\u001b[m\u001b[K->next = t);\n",
            "      |                    \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/sam_header.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Ksam_hdr_vadd\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/sam_header.c:449:16:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Ksuggest parentheses around assignment used as truth value [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wparentheses\u0007-Wparentheses\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  449 |         assert(\u001b[01;35m\u001b[Kp\u001b[m\u001b[K->next = t);\n",
            "      |                \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/string_alloc.o cram/string_alloc.c\n",
            "In file included from \u001b[01m\u001b[K/usr/include/string.h:535\u001b[m\u001b[K,\n",
            "                 from \u001b[01m\u001b[Kcram/string_alloc.c:39\u001b[m\u001b[K:\n",
            "In function ‘\u001b[01m\u001b[Kstrncpy\u001b[m\u001b[K’,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_ndup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:149:5\u001b[m\u001b[K,\n",
            "    inlined from ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’ at \u001b[01m\u001b[Kcram/string_alloc.c:141:12\u001b[m\u001b[K:\n",
            "\u001b[01m\u001b[K/usr/include/x86_64-linux-gnu/bits/string_fortified.h:95:10:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K__builtin_strncpy\u001b[m\u001b[K’ output truncated before terminating nul copying as many bytes from a string as its length [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wstringop-truncation\u0007-Wstringop-truncation\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   95 |   return \u001b[01;35m\u001b[K__builtin___strncpy_chk (__dest, __src, __len,\u001b[m\u001b[K\n",
            "      |          \u001b[01;35m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "   96 | \u001b[01;35m\u001b[K                                  __glibc_objsize (__dest))\u001b[m\u001b[K;\n",
            "      |                                   \u001b[01;35m\u001b[K~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kstring_dup\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/string_alloc.c:141:12:\u001b[m\u001b[K \u001b[01;36m\u001b[Knote: \u001b[m\u001b[Klength computed here\n",
            "  141 |     return \u001b[01;36m\u001b[Kstring_ndup(a_str, instr, strlen(instr))\u001b[m\u001b[K;\n",
            "      |            \u001b[01;36m\u001b[K^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/thread_pool.o cram/thread_pool.c\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/vlen.o cram/vlen.c\n",
            "\u001b[01m\u001b[Kcram/vlen.c:\u001b[m\u001b[K In function ‘\u001b[01m\u001b[Kvflen\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[Kcram/vlen.c:121:9:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Ki\u001b[m\u001b[K’ set but not used [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-but-set-variable\u0007-Wunused-but-set-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "  121 |     int \u001b[01;35m\u001b[Ki\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K^\u001b[m\u001b[K\n",
            "cc -g -Wall -O2  -I. -DSAMTOOLS=1 -c -o cram/zfio.o cram/zfio.c\n",
            "ar -rc libhts.a kfunc.o knetfile.o kstring.o bgzf.o faidx.o hfile.o hfile_net.o hts.o sam.o synced_bcf_reader.o vcf_sweep.o tbx.o vcf.o vcfutils.o cram/cram_codecs.o cram/cram_decode.o cram/cram_encode.o cram/cram_index.o cram/cram_io.o cram/cram_samtools.o cram/cram_stats.o cram/files.o cram/mFILE.o cram/md5.o cram/open_trace_file.o cram/pooled_alloc.o cram/sam_header.o cram/string_alloc.o cram/thread_pool.o cram/vlen.o cram/zfio.o\n",
            "ranlib libhts.a\n",
            "make[1]: Leaving directory '/content/STAR-2.7.10a/source/htslib'\n",
            "echo BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall_multiMappers.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c\n",
            "BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall_multiMappers.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c\n",
            "'rm' -f ./Depend.list\n",
            "g++ -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:14+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -MM BAMbinSortByCoordinate.cpp BAMbinSortUnmapped.cpp BAMfunctions.cpp BAMoutput.cpp bamRemoveDuplicates.cpp bamSortByCoordinate.cpp binarySearch2.cpp blocksOverlap.cpp Chain.cpp ChimericAlign_chimericBAMoutput.cpp ChimericAlign_chimericJunctionOutput.cpp ChimericAlign_chimericStitching.cpp ChimericAlign.cpp ChimericDetection_chimericDetectionMult.cpp ChimericDetection.cpp ChimericSegment.cpp ClipCR4.cpp ClipMate_clipChunk.cpp ClipMate_clip.cpp ClipMate_initialize.cpp ErrorWarning.cpp extendAlign.cpp funCompareUintAndSuffixes.cpp funCompareUintAndSuffixesMemcmp.cpp Genome.cpp Genome_genomeGenerate.cpp Genome_genomeLoad.cpp Genome_genomeOutLoad.cpp Genome_insertSequences.cpp genomeParametersWrite.cpp genomeSAindex.cpp genomeScanFastaFiles.cpp Genome_transformGenome.cpp GlobalVariables.cpp GTF.cpp GTF_superTranscript.cpp GTF_transcriptGeneSJ.cpp InOutStreams.cpp insertSeqSA.cpp mapThreadsSpawn.cpp outputSJ.cpp OutSJ.cpp PackedArray.cpp ParametersChimeric_initialize.cpp ParametersClip_initialize.cpp Parameters_closeReadsFiles.cpp Parameters.cpp ParametersGenome.cpp Parameters_openReadsFiles.cpp Parameters_readFilesInit.cpp Parameters_readSAMheader.cpp Parameters_samAttributes.cpp ParametersSolo.cpp Quantifications.cpp ReadAlign_alignBAM.cpp ReadAlign_assignAlignToWindow.cpp ReadAlign_calcCIGAR.cpp ReadAlign_chimericDetection.cpp ReadAlign_chimericDetectionOld.cpp ReadAlign_chimericDetectionOldOutput.cpp ReadAlign_chimericDetectionPEmerged.cpp ReadAlignChunk.cpp ReadAlignChunk_mapChunk.cpp ReadAlignChunk_processChunks.cpp ReadAlign_CIGAR.cpp ReadAlign.cpp ReadAlign_createExtendWindowsWithAlign.cpp ReadAlign_mapOneRead.cpp ReadAlign_mapOneReadSpliceGraph.cpp ReadAlign_mappedFilter.cpp ReadAlign_maxMappableLength2strands.cpp ReadAlign_multMapSelect.cpp ReadAlign_oneRead.cpp ReadAlign_outputAlignments.cpp ReadAlign_outputSpliceGraphSAM.cpp ReadAlign_outputTranscriptCIGARp.cpp ReadAlign_outputTranscriptSAM.cpp ReadAlign_outputTranscriptSJ.cpp ReadAlign_outputVariation.cpp ReadAlign_peOverlapMergeMap.cpp ReadAlign_quantTranscriptome.cpp ReadAlign_stitchPieces.cpp ReadAlign_stitchWindowSeeds.cpp ReadAlign_storeAligns.cpp ReadAlign_transformGenome.cpp ReadAlign_waspMap.cpp readLoad.cpp samHeaders.cpp SequenceFuns.cpp serviceFuns.cpp SharedMemory.cpp signalFromBAM.cpp sjAlignSplit.cpp sjdbBuildIndex.cpp sjdbInsertJunctions.cpp sjdbLoadFromFiles.cpp sjdbLoadFromStream.cpp sjdbPrepare.cpp SoloBarcode.cpp SoloBarcode_extractBarcode.cpp Solo.cpp SoloFeature_addBAMtags.cpp SoloFeature_cellFiltering.cpp SoloFeature_collapseUMIall_multiMappers.cpp SoloFeature_collapseUMI_Graph.cpp SoloFeature_countCBgeneUMI.cpp SoloFeature_countSmartSeq.cpp SoloFeature_countVelocyto.cpp SoloFeature.cpp SoloFeature_emptyDrops_CR.cpp SoloFeature_loadRawMatrix.cpp SoloFeature_outputResults.cpp SoloFeature_processRecords.cpp SoloFeature_quantTranscript.cpp SoloFeature_redistributeReadsByCB.cpp SoloFeature_statsOutput.cpp SoloFeature_sumThreads.cpp soloInputFeatureUMI.cpp SoloReadBarcode.cpp SoloReadBarcode_getCBandUMI.cpp SoloRead.cpp SoloReadFeature.cpp SoloReadFeature_inputRecords.cpp SoloReadFeature_record.cpp SoloRead_record.cpp SpliceGraph.cpp SpliceGraph_findSuperTr.cpp SpliceGraph_swScoreSpliced.cpp SpliceGraph_swTraceBack.cpp STAR.cpp Stats.cpp stitchAlignToTranscript.cpp stitchGapIndel.cpp stitchWindowAligns.cpp streamFuns.cpp stringSubstituteAll.cpp SuffixArrayFuns.cpp SuperTranscriptome.cpp sysRemoveDir.cpp ThreadControl.cpp TimeFunctions.cpp Transcript_alignScore.cpp Transcript_convertGenomeCigar.cpp Transcript.cpp Transcript_generateCigarP.cpp Transcriptome_alignExonOverlap.cpp Transcriptome_classifyAlign.cpp Transcriptome.cpp Transcriptome_geneCountsAddAlign.cpp Transcriptome_geneFullAlignOverlap.cpp Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp Transcriptome_quantAlign.cpp Transcript_transformGenome.cpp Transcript_variationAdjust.cpp Transcript_variationOutput.cpp twoPassRunPass1.cpp Variation.cpp bam_cat.c parametersDefault.xxd htslib >> Depend.list\n",
            "\u001b[01m\u001b[Kg++:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[KparametersDefault.xxd: linker input file unused because linking not done\n",
            "\u001b[01m\u001b[Kg++:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Khtslib: linker input file unused because linking not done\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_collapseUMI_Graph.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_collapseUMIall_multiMappers.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ParametersClip_initialize.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ClipMate_clip.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ClipCR4.cpp\n",
            "cd opal && \\\n",
            "g++ -c -I./ -std=c++11   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra  -mavx2 opal.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ClipMate_clipChunk.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ClipMate_initialize.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_loadRawMatrix.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_emptyDrops_CR.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra soloInputFeatureUMI.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_countSmartSeq.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_redistributeReadsByCB.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_quantTranscript.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_sumThreads.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_countVelocyto.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_countCBgeneUMI.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_classifyAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_geneFullAlignOverlap_ExonOverIntron.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_cellFiltering.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_statsOutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra bamSortByCoordinate.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloBarcode.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ParametersSolo.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloRead.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloRead_record.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadBarcode.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadBarcode_getCBandUMI.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloBarcode_extractBarcode.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadFeature.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadFeature_record.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloReadFeature_inputRecords.cpp\n",
            "\u001b[01m\u001b[KSoloReadFeature_inputRecords.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid SoloReadFeature::inputRecords(unsigned int**, unsigned int, std::vector<unsigned int>&, std::vector<readInfoStruct>&, SoloReadFlagClass&, std::vector<unsigned int>&, std::vector<unsigned int>&)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KSoloReadFeature_inputRecords.cpp:21:12:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kunused variable ‘\u001b[01m\u001b[KnReadsIn\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-variable\u0007-Wunused-variable\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   21 |     uint64 \u001b[01;35m\u001b[KnReadsIn\u001b[m\u001b[K = 0;\n",
            "      |            \u001b[01;35m\u001b[K^~~~~~~~\u001b[m\u001b[K\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Solo.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_outputResults.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_processRecords.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_addBAMtags.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_transformGenome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_transformGenome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_convertGenomeCigar.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra twoPassRunPass1.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra samHeaders.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_genomeLoad.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_genomeOutLoad.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_transformGenome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputSpliceGraphSAM.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_mapOneReadSpliceGraph.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SpliceGraph.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SpliceGraph_swScoreSpliced.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SpliceGraph_swTraceBack.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SpliceGraph_findSuperTr.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjAlignSplit.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra GTF.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra GTF_transcriptGeneSJ.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra GTF_superTranscript.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SuperTranscriptome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputAlignments.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra STAR.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SharedMemory.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra PackedArray.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SuffixArrayFuns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Parameters.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Parameters_samAttributes.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra InOutStreams.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SequenceFuns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ParametersGenome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Stats.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_alignScore.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_generateCigarP.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Chain.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcript_variationAdjust.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Variation.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_waspMap.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_storeAligns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_stitchPieces.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_multMapSelect.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_mapOneRead.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra readLoad.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlignChunk.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlignChunk_processChunks.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlignChunk_mapChunk.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra OutSJ.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra outputSJ.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra blocksOverlap.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ThreadControl.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sysRemoveDir.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_maxMappableLength2strands.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra binarySearch2.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputTranscriptSAM.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputTranscriptSJ.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_outputTranscriptCIGARp.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_createExtendWindowsWithAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_assignAlignToWindow.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_oneRead.cpp\n",
            "\u001b[01m\u001b[KReadAlign_oneRead.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kint ReadAlign::oneRead()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KReadAlign_oneRead.cpp:23:21:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[K‘\u001b[01m\u001b[K*readStatus[0]\u001b[m\u001b[K’ may be used uninitialized [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wmaybe-uninitialized\u0007-Wmaybe-uninitialized\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   23 |     if (\u001b[01;35m\u001b[KreadStatus[0]\u001b[m\u001b[K==-1) {//finished with the stream\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~~~~~~~^\u001b[m\u001b[K\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_stitchWindowSeeds.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_peOverlapMergeMap.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_mappedFilter.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ParametersChimeric_initialize.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_chimericDetection.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_chimericDetectionOld.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_chimericDetectionOldOutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericDetection.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericDetection_chimericDetectionMult.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_chimericDetectionPEmerged.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra stitchWindowAligns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra extendAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra stitchAlignToTranscript.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericAlign_chimericJunctionOutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericAlign_chimericBAMoutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ChimericAlign_chimericStitching.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_genomeGenerate.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra genomeParametersWrite.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra genomeScanFastaFiles.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra genomeSAindex.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Genome_insertSequences.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra insertSeqSA.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra funCompareUintAndSuffixes.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra funCompareUintAndSuffixesMemcmp.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra TimeFunctions.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ErrorWarning.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra streamFuns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra stringSubstituteAll.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_quantAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_geneFullAlignOverlap.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_quantTranscriptome.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Quantifications.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Transcriptome_geneCountsAddAlign.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbLoadFromFiles.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbLoadFromStream.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbPrepare.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbBuildIndex.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra sjdbInsertJunctions.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra mapThreadsSpawn.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Parameters_readFilesInit.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra Parameters_readSAMheader.cpp\n",
            "\u001b[01m\u001b[KParameters_readSAMheader.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid Parameters::readSAMheader(std::string, std::vector<std::__cxx11::basic_string<char> >)\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KParameters_readSAMheader.cpp:32:15:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kignoring return value of ‘\u001b[01m\u001b[Kint system(const char*)\u001b[m\u001b[K’ declared with attribute ‘\u001b[01m\u001b[Kwarn_unused_result\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wunused-result\u0007-Wunused-result\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   32 |         \u001b[01;35m\u001b[Ksystem(com1.c_str())\u001b[m\u001b[K;\n",
            "      |         \u001b[01;35m\u001b[K~~~~~~^~~~~~~~~~~~~~\u001b[m\u001b[K\n",
            "g++ -c  -pipe -Wall -Wextra -O3 bam_cat.c\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra serviceFuns.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra BAMoutput.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra BAMfunctions.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra ReadAlign_alignBAM.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra BAMbinSortByCoordinate.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra signalFromBAM.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra bamRemoveDuplicates.cpp\n",
            "g++ -c   -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra BAMbinSortUnmapped.cpp\n",
            "g++ -o STAR  -O3 -std=c++11 -fopenmp -D'COMPILATION_TIME_PLACE=\"2025-11-11T07:05:51+00:00 07292e4ac9e7:/content/STAR-2.7.10a/source\"' -D'GIT_BRANCH_COMMIT_DIFF=\"\"' -pipe -Wall -Wextra SoloFeature_collapseUMI_Graph.o SoloFeature_collapseUMIall_multiMappers.o ParametersClip_initialize.o ClipMate_clip.o ClipCR4.o opal/opal.o ClipMate_clipChunk.o ClipMate_initialize.o SoloFeature_loadRawMatrix.o SoloFeature_emptyDrops_CR.o soloInputFeatureUMI.o SoloFeature_countSmartSeq.o SoloFeature_redistributeReadsByCB.o SoloFeature_quantTranscript.o SoloFeature_sumThreads.o SoloFeature_countVelocyto.o SoloFeature_countCBgeneUMI.o Transcriptome_classifyAlign.o Transcriptome_geneFullAlignOverlap_ExonOverIntron.o Transcriptome_alignExonOverlap.cpp SoloFeature_cellFiltering.o SoloFeature_statsOutput.o bamSortByCoordinate.o SoloBarcode.o ParametersSolo.o SoloRead.o SoloRead_record.o SoloReadBarcode.o SoloReadBarcode_getCBandUMI.o SoloBarcode_extractBarcode.o SoloReadFeature.o SoloReadFeature_record.o SoloReadFeature_inputRecords.o Solo.o SoloFeature.o SoloFeature_outputResults.o SoloFeature_processRecords.o SoloFeature_addBAMtags.o ReadAlign_transformGenome.o Genome_transformGenome.o Transcript_convertGenomeCigar.o twoPassRunPass1.o samHeaders.o Genome_genomeLoad.o Genome_genomeOutLoad.o Transcript_transformGenome.o ReadAlign_outputSpliceGraphSAM.o ReadAlign_mapOneReadSpliceGraph.o SpliceGraph.o SpliceGraph_swScoreSpliced.o SpliceGraph_swTraceBack.o SpliceGraph_findSuperTr.o sjAlignSplit.o GTF.o GTF_transcriptGeneSJ.o GTF_superTranscript.o SuperTranscriptome.o ReadAlign_outputAlignments.o ReadAlign.o STAR.o SharedMemory.o PackedArray.o SuffixArrayFuns.o Parameters.o Parameters_samAttributes.o InOutStreams.o SequenceFuns.o Genome.o ParametersGenome.o Stats.o Transcript.o Transcript_alignScore.o Transcript_generateCigarP.o Chain.o Transcript_variationAdjust.o Variation.o ReadAlign_waspMap.o ReadAlign_storeAligns.o ReadAlign_stitchPieces.o ReadAlign_multMapSelect.o ReadAlign_mapOneRead.o readLoad.o ReadAlignChunk.o ReadAlignChunk_processChunks.o ReadAlignChunk_mapChunk.o OutSJ.o outputSJ.o blocksOverlap.o ThreadControl.o sysRemoveDir.o ReadAlign_maxMappableLength2strands.o binarySearch2.o ReadAlign_outputTranscriptSAM.o ReadAlign_outputTranscriptSJ.o ReadAlign_outputTranscriptCIGARp.o ReadAlign_calcCIGAR.cpp ReadAlign_createExtendWindowsWithAlign.o ReadAlign_assignAlignToWindow.o ReadAlign_oneRead.o ReadAlign_stitchWindowSeeds.o ReadAlign_peOverlapMergeMap.o ReadAlign_mappedFilter.o ParametersChimeric_initialize.o ReadAlign_chimericDetection.o ReadAlign_chimericDetectionOld.o ReadAlign_chimericDetectionOldOutput.o ChimericDetection.o ChimericDetection_chimericDetectionMult.o ReadAlign_chimericDetectionPEmerged.o stitchWindowAligns.o extendAlign.o stitchAlignToTranscript.o ChimericSegment.cpp ChimericAlign.cpp ChimericAlign_chimericJunctionOutput.o ChimericAlign_chimericBAMoutput.o ChimericAlign_chimericStitching.o Genome_genomeGenerate.o genomeParametersWrite.o genomeScanFastaFiles.o genomeSAindex.o Genome_insertSequences.o insertSeqSA.o funCompareUintAndSuffixes.o funCompareUintAndSuffixesMemcmp.o TimeFunctions.o ErrorWarning.o streamFuns.o stringSubstituteAll.o Transcriptome.o Transcriptome_quantAlign.o Transcriptome_geneFullAlignOverlap.o ReadAlign_quantTranscriptome.o Quantifications.o Transcriptome_geneCountsAddAlign.o sjdbLoadFromFiles.o sjdbLoadFromStream.o sjdbPrepare.o sjdbBuildIndex.o sjdbInsertJunctions.o mapThreadsSpawn.o Parameters_readFilesInit.o Parameters_openReadsFiles.cpp Parameters_closeReadsFiles.cpp Parameters_readSAMheader.o bam_cat.o serviceFuns.o GlobalVariables.cpp BAMoutput.o BAMfunctions.o ReadAlign_alignBAM.o BAMbinSortByCoordinate.o signalFromBAM.o bamRemoveDuplicates.o BAMbinSortUnmapped.o  -pthread -Lhtslib -Bstatic -lhts -Bdynamic -lz \n",
            "\u001b[01m\u001b[KParameters_openReadsFiles.cpp:\u001b[m\u001b[K In member function ‘\u001b[01m\u001b[Kvoid Parameters::openReadsFiles()\u001b[m\u001b[K’:\n",
            "\u001b[01m\u001b[KParameters_openReadsFiles.cpp:25:20:\u001b[m\u001b[K \u001b[01;35m\u001b[Kwarning: \u001b[m\u001b[Kvariable ‘\u001b[01m\u001b[Kimate\u001b[m\u001b[K’ might be clobbered by ‘\u001b[01m\u001b[Klongjmp\u001b[m\u001b[K’ or ‘\u001b[01m\u001b[Kvfork\u001b[m\u001b[K’ [\u001b[01;35m\u001b[K\u001b]8;;https://gcc.gnu.org/onlinedocs/gcc/Warning-Options.html#index-Wclobbered\u0007-Wclobbered\u001b]8;;\u0007\u001b[m\u001b[K]\n",
            "   25 |          for (uint \u001b[01;35m\u001b[Kimate\u001b[m\u001b[K=0;imate<readFilesNames.size();imate++) {//open readIn files\n",
            "      |                    \u001b[01;35m\u001b[K^~~~~\u001b[m\u001b[K\n",
            "bin\t\t    _config.yml      docs     README.md\n",
            "CHANGES.md\t    CONTRIBUTING.md  extras   RELEASEnotes.md\n",
            "CODE_OF_CONDUCT.md  doc\t\t     LICENSE  source\n"
          ]
        }
      ],
      "source": [
        "# Step 0: Install Required Tools and Libraries\n",
        "# Update package list\n",
        "!apt-get update\n",
        "\n",
        "# Install build tools and SRA Toolkit\n",
        "!apt-get install -y build-essential sra-toolkit unzip\n",
        "\n",
        "# Download and compile STAR\n",
        "!wget https://github.com/alexdobin/STAR/archive/2.7.10a.zip\n",
        "!unzip 2.7.10a.zip\n",
        "!ls STAR-2.7.10a/ # List contents after unzip\n",
        "!cd STAR-2.7.10a/source && make STAR\n",
        "!ls STAR-2.7.10a/ # List contents after make"
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Step 1: Download FASTQ Files from SRA\n",
        "# Download and convert SRR11548957 to paired-end FASTQ files\n",
        "!prefetch SRR15509125\n",
        "!fasterq-dump SRR15509125 --split-files\n",
        "!echo $? # Check the exit status of the fasterq-dump command\n",
        "!ls # List files after fasterq-dump to see if they were created"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "RIvx6ie3kaER",
        "outputId": "ba4daae1-d042-4fd9-9b0d-f3bb916f72d7"
      },
      "execution_count": 5,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\n",
            "2025-11-11T07:09:59 prefetch.2.11.3: Current preference is set to retrieve SRA Normalized Format files with full base quality scores.\n",
            "2025-11-11T07:09:59 prefetch.2.11.3: 1) Downloading 'SRR15509125'...\n",
            "2025-11-11T07:09:59 prefetch.2.11.3: SRA Normalized Format file is being retrieved, if this is different from your preference, it may be due to current file availability.\n",
            "2025-11-11T07:09:59 prefetch.2.11.3:  Downloading via HTTPS...\n",
            "2025-11-11T07:10:02 prefetch.2.11.3:  HTTPS download succeed\n",
            "2025-11-11T07:10:03 prefetch.2.11.3:  'SRR15509125' is valid\n",
            "2025-11-11T07:10:03 prefetch.2.11.3: 1) 'SRR15509125' was downloaded successfully\n",
            "2025-11-11T07:10:03 prefetch.2.11.3: 'SRR15509125' has 0 unresolved dependencies\n",
            "spots read      : 407,248\n",
            "reads read      : 814,496\n",
            "reads written   : 814,496\n",
            "0\n",
            "2.7.10a.zip  SRR15509125\t  SRR15509125_2.fastq  star_output\n",
            "sample_data  SRR15509125_1.fastq  STAR-2.7.10a\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Step 2: Prepare Reference Genome (chr22)\n",
        "# Create reference directory and download chr22 from Ensembl\n",
        "!mkdir -p reference\n",
        "!wget -P reference ftp://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n",
        "!wget -P reference ftp://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz\n",
        "\n",
        "# Unzip the FASTA and GTF files\n",
        "!gunzip reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n",
        "!gunzip reference/Homo_sapiens.GRCh38.109.gtf.gz"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "mNnlqnddkjg_",
        "outputId": "00253b99-b5a7-4d5d-9218-d79e604ec2a8"
      },
      "execution_count": 6,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "--2025-11-11 07:10:38--  ftp://ftp.ensembl.org/pub/release-109/fasta/homo_sapiens/dna/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz\n",
            "           => ‘reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz’\n",
            "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
            "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /pub/release-109/fasta/homo_sapiens/dna ... done.\n",
            "==> SIZE Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz ... 11389810\n",
            "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz ... done.\n",
            "Length: 11389810 (11M) (unauthoritative)\n",
            "\n",
            "Homo_sapiens.GRCh38 100%[===================>]  10.86M  10.1MB/s    in 1.1s    \n",
            "\n",
            "2025-11-11 07:10:41 (10.1 MB/s) - ‘reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa.gz’ saved [11389810]\n",
            "\n",
            "--2025-11-11 07:10:41--  ftp://ftp.ensembl.org/pub/release-109/gtf/homo_sapiens/Homo_sapiens.GRCh38.109.gtf.gz\n",
            "           => ‘reference/Homo_sapiens.GRCh38.109.gtf.gz’\n",
            "Resolving ftp.ensembl.org (ftp.ensembl.org)... 193.62.193.169\n",
            "Connecting to ftp.ensembl.org (ftp.ensembl.org)|193.62.193.169|:21... connected.\n",
            "Logging in as anonymous ... Logged in!\n",
            "==> SYST ... done.    ==> PWD ... done.\n",
            "==> TYPE I ... done.  ==> CWD (1) /pub/release-109/gtf/homo_sapiens ... done.\n",
            "==> SIZE Homo_sapiens.GRCh38.109.gtf.gz ... 54258835\n",
            "==> PASV ... done.    ==> RETR Homo_sapiens.GRCh38.109.gtf.gz ... done.\n",
            "Length: 54258835 (52M) (unauthoritative)\n",
            "\n",
            "Homo_sapiens.GRCh38 100%[===================>]  51.75M  21.9MB/s    in 2.4s    \n",
            "\n",
            "2025-11-11 07:10:45 (21.9 MB/s) - ‘reference/Homo_sapiens.GRCh38.109.gtf.gz’ saved [54258835]\n",
            "\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#  Step 3: Generate STAR Genome Index\n",
        "# Create index directory and build genome index\n",
        "!mkdir -p star_index\n",
        "!/content/STAR-2.7.10a/bin/Linux_x86_64/STAR \\\n",
        "    --runThreadN 4 \\\n",
        "    --runMode genomeGenerate \\\n",
        "    --genomeDir star_index \\\n",
        "    --genomeFastaFiles reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa \\\n",
        "    --sjdbGTFfile reference/Homo_sapiens.GRCh38.109.gtf \\\n",
        "    --genomeSAindexNbases 9"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "bHe0Dhw1koJd",
        "outputId": "c756a4be-ce78-4c1c-882d-a357e8bd6da1"
      },
      "execution_count": 7,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\t/content/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 4 --runMode genomeGenerate --genomeDir star_index --genomeFastaFiles reference/Homo_sapiens.GRCh38.dna.chromosome.22.fa --sjdbGTFfile reference/Homo_sapiens.GRCh38.109.gtf --genomeSAindexNbases 9\n",
            "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
            "Nov 11 07:11:08 ..... started STAR run\n",
            "Nov 11 07:11:08 ... starting to generate Genome files\n",
            "Nov 11 07:11:10 ..... processing annotations GTF\n",
            "Nov 11 07:11:23 ... starting to sort Suffix Array. This may take a long time...\n",
            "Nov 11 07:11:23 ... sorting Suffix Array chunks and saving them to disk...\n",
            "Nov 11 07:12:16 ... loading chunks from disk, packing SA...\n",
            "Nov 11 07:12:18 ... finished generating suffix array\n",
            "Nov 11 07:12:18 ... generating Suffix Array index\n",
            "Nov 11 07:12:19 ... completed Suffix Array index\n",
            "Nov 11 07:12:19 ..... inserting junctions into the genome indices\n",
            "Nov 11 07:12:30 ... writing Genome to disk ...\n",
            "Nov 11 07:12:30 ... writing Suffix Array to disk ...\n",
            "Nov 11 07:12:34 ... writing SAindex to disk\n",
            "Nov 11 07:12:34 ..... finished successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "c8f78677",
        "outputId": "dc5a142b-2e6c-49df-c3b9-5688e49ec84b"
      },
      "source": [
        "# Verify that the FASTQ files were created\n",
        "!ls"
      ],
      "execution_count": 8,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "2.7.10a.zip  sample_data  SRR15509125_1.fastq  STAR-2.7.10a  star_output\n",
            "reference    SRR15509125  SRR15509125_2.fastq  star_index\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#  Step 4: Run STAR Alignment\n",
        "# Create output directory and align reads to reference genome\n",
        "!mkdir -p star_output\n",
        "!/content/STAR-2.7.10a/bin/Linux_x86_64/STAR \\\n",
        "    --runThreadN 4 \\\n",
        "    --genomeDir star_index \\\n",
        "    --readFilesIn SRR15509125_1.fastq SRR15509125_2.fastq \\\n",
        "    --outFileNamePrefix star_output/SRR15509125_ \\\n",
        "    --outSAMtype BAM SortedByCoordinate \\\n",
        "    --quantMode GeneCounts"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "juEIvPVckrvI",
        "outputId": "0fd49a05-e537-4148-d377-4cb2ddaf3bc9"
      },
      "execution_count": 9,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "\t/content/STAR-2.7.10a/bin/Linux_x86_64/STAR --runThreadN 4 --genomeDir star_index --readFilesIn SRR15509125_1.fastq SRR15509125_2.fastq --outFileNamePrefix star_output/SRR15509125_ --outSAMtype BAM SortedByCoordinate --quantMode GeneCounts\n",
            "\tSTAR version: 2.7.10a   compiled: 2022-01-14T18:50:00-05:00 :/home/dobin/data/STAR/STARcode/STAR.master/source\n",
            "Nov 11 07:12:43 ..... started STAR run\n",
            "Nov 11 07:12:43 ..... loading genome\n",
            "Nov 11 07:12:43 ..... started mapping\n",
            "Nov 11 08:33:19 ..... finished mapping\n",
            "Nov 11 08:33:20 ..... started sorting BAM\n",
            "Nov 11 08:33:20 ..... finished successfully\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "#  Step 5: Inspect Output Files\n",
        "# List key output files\n",
        "!ls -lh star_output/SRR15509125_Aligned.sortedByCoord.out.bam\n",
        "!ls -lh star_output/SRR15509125_ReadsPerGene.out.tab\n",
        "!cat star_output/SRR15509125_Log.final.out\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/"
        },
        "id": "DnWncBHeFJvp",
        "outputId": "4cb14272-1a7d-46e0-b61e-0c7bf2c3ec5a"
      },
      "execution_count": 13,
      "outputs": [
        {
          "output_type": "stream",
          "name": "stdout",
          "text": [
            "-rw-r--r-- 1 root root 1.3K Nov 11 08:33 star_output/SRR15509125_Aligned.sortedByCoord.out.bam\n",
            "-rw-r--r-- 1 root root 32K Nov 11 08:33 star_output/SRR15509125_ReadsPerGene.out.tab\n",
            "                                 Started job on |\tNov 11 07:12:43\n",
            "                             Started mapping on |\tNov 11 07:12:43\n",
            "                                    Finished on |\tNov 11 08:33:20\n",
            "       Mapping speed, Million of reads per hour |\t0.30\n",
            "\n",
            "                          Number of input reads |\t407248\n",
            "                      Average input read length |\t458\n",
            "                                    UNIQUE READS:\n",
            "                   Uniquely mapped reads number |\t7\n",
            "                        Uniquely mapped reads % |\t0.00%\n",
            "                          Average mapped length |\t64.29\n",
            "                       Number of splices: Total |\t18\n",
            "            Number of splices: Annotated (sjdb) |\t0\n",
            "                       Number of splices: GT/AG |\t14\n",
            "                       Number of splices: GC/AG |\t0\n",
            "                       Number of splices: AT/AC |\t0\n",
            "               Number of splices: Non-canonical |\t4\n",
            "                      Mismatch rate per base, % |\t5.33%\n",
            "                         Deletion rate per base |\t0.00%\n",
            "                        Deletion average length |\t0.00\n",
            "                        Insertion rate per base |\t0.00%\n",
            "                       Insertion average length |\t0.00\n",
            "                             MULTI-MAPPING READS:\n",
            "        Number of reads mapped to multiple loci |\t0\n",
            "             % of reads mapped to multiple loci |\t0.00%\n",
            "        Number of reads mapped to too many loci |\t0\n",
            "             % of reads mapped to too many loci |\t0.00%\n",
            "                                  UNMAPPED READS:\n",
            "  Number of reads unmapped: too many mismatches |\t0\n",
            "       % of reads unmapped: too many mismatches |\t0.00%\n",
            "            Number of reads unmapped: too short |\t196358\n",
            "                 % of reads unmapped: too short |\t48.22%\n",
            "                Number of reads unmapped: other |\t210883\n",
            "                     % of reads unmapped: other |\t51.78%\n",
            "                                  CHIMERIC READS:\n",
            "                       Number of chimeric reads |\t0\n",
            "                            % of chimeric reads |\t0.00%\n"
          ]
        }
      ]
    },
    {
      "cell_type": "code",
      "source": [
        "# Step 6: Download Output Files (Colab)\n",
        "from google.colab import files\n",
        "files.download('star_output/SRR15509125_Aligned.sortedByCoord.out.bam')\n",
        "files.download('star_output/SRR15509125_ReadsPerGene.out.tab')\n",
        "files.download('star_output/SRR15509125_Log.final.out')\n"
      ],
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 17
        },
        "id": "zvsZM6p_G8yG",
        "outputId": "c7088a49-992e-4b52-a094-71464790cd6b"
      },
      "execution_count": 17,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_83d82b76-e0c2-4c21-aef8-22bcf3e9d029\", \"SRR15509125_Aligned.sortedByCoord.out.bam\", 1296)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_72eea39d-da0e-4cd3-89c8-2dd42397df02\", \"SRR15509125_ReadsPerGene.out.tab\", 31879)"
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "\n",
              "    async function download(id, filename, size) {\n",
              "      if (!google.colab.kernel.accessAllowed) {\n",
              "        return;\n",
              "      }\n",
              "      const div = document.createElement('div');\n",
              "      const label = document.createElement('label');\n",
              "      label.textContent = `Downloading \"${filename}\": `;\n",
              "      div.appendChild(label);\n",
              "      const progress = document.createElement('progress');\n",
              "      progress.max = size;\n",
              "      div.appendChild(progress);\n",
              "      document.body.appendChild(div);\n",
              "\n",
              "      const buffers = [];\n",
              "      let downloaded = 0;\n",
              "\n",
              "      const channel = await google.colab.kernel.comms.open(id);\n",
              "      // Send a message to notify the kernel that we're ready.\n",
              "      channel.send({})\n",
              "\n",
              "      for await (const message of channel.messages) {\n",
              "        // Send a message to notify the kernel that we're ready.\n",
              "        channel.send({})\n",
              "        if (message.buffers) {\n",
              "          for (const buffer of message.buffers) {\n",
              "            buffers.push(buffer);\n",
              "            downloaded += buffer.byteLength;\n",
              "            progress.value = downloaded;\n",
              "          }\n",
              "        }\n",
              "      }\n",
              "      const blob = new Blob(buffers, {type: 'application/binary'});\n",
              "      const a = document.createElement('a');\n",
              "      a.href = window.URL.createObjectURL(blob);\n",
              "      a.download = filename;\n",
              "      div.appendChild(a);\n",
              "      a.click();\n",
              "      div.remove();\n",
              "    }\n",
              "  "
            ]
          },
          "metadata": {}
        },
        {
          "output_type": "display_data",
          "data": {
            "text/plain": [
              "<IPython.core.display.Javascript object>"
            ],
            "application/javascript": [
              "download(\"download_7c2f5274-c56c-4354-838b-9aa0cbcbb297\", \"SRR15509125_Log.final.out\", 1981)"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "code",
      "metadata": {
        "colab": {
          "base_uri": "https://localhost:8080/",
          "height": 542
        },
        "id": "a3ddb702",
        "outputId": "8ffa2c8c-d6d7-4e73-d250-71d25885860b"
      },
      "source": [
        "import plotly.graph_objects as go\n",
        "\n",
        "# Parse the STAR log file to extract mapping statistics\n",
        "log_file = 'star_output/SRR15509125_Log.final.out'\n",
        "mapping_stats = {}\n",
        "with open(log_file, 'r') as f:\n",
        "    for line in f:\n",
        "        if '%' in line:\n",
        "            parts = line.strip().split('|')\n",
        "            metric = parts[0].strip()\n",
        "            value = float(parts[1].strip().replace('%', ''))\n",
        "            mapping_stats[metric] = value\n",
        "\n",
        "# Define categories and values for plotting\n",
        "categories = [\n",
        "    'Uniquely mapped reads %',\n",
        "    '% of reads mapped to multiple loci',\n",
        "    '% of reads mapped to too many loci',\n",
        "    '% of reads unmapped: too short',\n",
        "    '% of reads unmapped: other',\n",
        "    '% of chimeric reads'\n",
        "]\n",
        "values = [mapping_stats.get(cat, 0) for cat in categories]\n",
        "\n",
        "# Create a Plotly bar chart\n",
        "fig = go.Figure(data=[go.Bar(x=categories, y=values)])\n",
        "\n",
        "# Update layout for better readability\n",
        "fig.update_layout(\n",
        "    title='STAR Alignment Summary',\n",
        "    xaxis_title='Read Categories',\n",
        "    yaxis_title='Percentage (%)'\n",
        ")\n",
        "\n",
        "# Display the plot\n",
        "fig.show()"
      ],
      "execution_count": 14,
      "outputs": [
        {
          "output_type": "display_data",
          "data": {
            "text/html": [
              "<html>\n",
              "<head><meta charset=\"utf-8\" /></head>\n",
              "<body>\n",
              "    <div>            <script src=\"https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js?config=TeX-AMS-MML_SVG\"></script><script type=\"text/javascript\">if (window.MathJax && window.MathJax.Hub && window.MathJax.Hub.Config) {window.MathJax.Hub.Config({SVG: {font: \"STIX-Web\"}});}</script>                <script type=\"text/javascript\">window.PlotlyConfig = {MathJaxConfig: 'local'};</script>\n",
              "        <script charset=\"utf-8\" src=\"https://cdn.plot.ly/plotly-2.35.2.min.js\"></script>                <div id=\"f03e518e-066e-427a-88ea-e071c7f49d44\" class=\"plotly-graph-div\" style=\"height:525px; width:100%;\"></div>            <script type=\"text/javascript\">                                    window.PLOTLYENV=window.PLOTLYENV || {};                                    if (document.getElementById(\"f03e518e-066e-427a-88ea-e071c7f49d44\")) {                    Plotly.newPlot(                        \"f03e518e-066e-427a-88ea-e071c7f49d44\",                        [{\"x\":[\"Uniquely mapped reads %\",\"% of reads mapped to multiple loci\",\"% of reads mapped to too many loci\",\"% of reads unmapped: too short\",\"% of reads unmapped: other\",\"% of chimeric reads\"],\"y\":[0.0,0.0,0.0,48.22,51.78,0.0],\"type\":\"bar\"}],                        {\"template\":{\"data\":{\"histogram2dcontour\":[{\"type\":\"histogram2dcontour\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"choropleth\":[{\"type\":\"choropleth\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"histogram2d\":[{\"type\":\"histogram2d\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"heatmap\":[{\"type\":\"heatmap\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"heatmapgl\":[{\"type\":\"heatmapgl\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"contourcarpet\":[{\"type\":\"contourcarpet\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"contour\":[{\"type\":\"contour\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"surface\":[{\"type\":\"surface\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"},\"colorscale\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]]}],\"mesh3d\":[{\"type\":\"mesh3d\",\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}],\"scatter\":[{\"fillpattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2},\"type\":\"scatter\"}],\"parcoords\":[{\"type\":\"parcoords\",\"line\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterpolargl\":[{\"type\":\"scatterpolargl\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"bar\":[{\"error_x\":{\"color\":\"#2a3f5f\"},\"error_y\":{\"color\":\"#2a3f5f\"},\"marker\":{\"line\":{\"color\":\"#E5ECF6\",\"width\":0.5},\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"bar\"}],\"scattergeo\":[{\"type\":\"scattergeo\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterpolar\":[{\"type\":\"scatterpolar\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"histogram\":[{\"marker\":{\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"histogram\"}],\"scattergl\":[{\"type\":\"scattergl\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatter3d\":[{\"type\":\"scatter3d\",\"line\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scattermapbox\":[{\"type\":\"scattermapbox\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scatterternary\":[{\"type\":\"scatterternary\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"scattercarpet\":[{\"type\":\"scattercarpet\",\"marker\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}}}],\"carpet\":[{\"aaxis\":{\"endlinecolor\":\"#2a3f5f\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"minorgridcolor\":\"white\",\"startlinecolor\":\"#2a3f5f\"},\"baxis\":{\"endlinecolor\":\"#2a3f5f\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"minorgridcolor\":\"white\",\"startlinecolor\":\"#2a3f5f\"},\"type\":\"carpet\"}],\"table\":[{\"cells\":{\"fill\":{\"color\":\"#EBF0F8\"},\"line\":{\"color\":\"white\"}},\"header\":{\"fill\":{\"color\":\"#C8D4E3\"},\"line\":{\"color\":\"white\"}},\"type\":\"table\"}],\"barpolar\":[{\"marker\":{\"line\":{\"color\":\"#E5ECF6\",\"width\":0.5},\"pattern\":{\"fillmode\":\"overlay\",\"size\":10,\"solidity\":0.2}},\"type\":\"barpolar\"}],\"pie\":[{\"automargin\":true,\"type\":\"pie\"}]},\"layout\":{\"autotypenumbers\":\"strict\",\"colorway\":[\"#636efa\",\"#EF553B\",\"#00cc96\",\"#ab63fa\",\"#FFA15A\",\"#19d3f3\",\"#FF6692\",\"#B6E880\",\"#FF97FF\",\"#FECB52\"],\"font\":{\"color\":\"#2a3f5f\"},\"hovermode\":\"closest\",\"hoverlabel\":{\"align\":\"left\"},\"paper_bgcolor\":\"white\",\"plot_bgcolor\":\"#E5ECF6\",\"polar\":{\"bgcolor\":\"#E5ECF6\",\"angularaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"radialaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"}},\"ternary\":{\"bgcolor\":\"#E5ECF6\",\"aaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"baxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"},\"caxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\"}},\"coloraxis\":{\"colorbar\":{\"outlinewidth\":0,\"ticks\":\"\"}},\"colorscale\":{\"sequential\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"sequentialminus\":[[0.0,\"#0d0887\"],[0.1111111111111111,\"#46039f\"],[0.2222222222222222,\"#7201a8\"],[0.3333333333333333,\"#9c179e\"],[0.4444444444444444,\"#bd3786\"],[0.5555555555555556,\"#d8576b\"],[0.6666666666666666,\"#ed7953\"],[0.7777777777777778,\"#fb9f3a\"],[0.8888888888888888,\"#fdca26\"],[1.0,\"#f0f921\"]],\"diverging\":[[0,\"#8e0152\"],[0.1,\"#c51b7d\"],[0.2,\"#de77ae\"],[0.3,\"#f1b6da\"],[0.4,\"#fde0ef\"],[0.5,\"#f7f7f7\"],[0.6,\"#e6f5d0\"],[0.7,\"#b8e186\"],[0.8,\"#7fbc41\"],[0.9,\"#4d9221\"],[1,\"#276419\"]]},\"xaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\",\"title\":{\"standoff\":15},\"zerolinecolor\":\"white\",\"automargin\":true,\"zerolinewidth\":2},\"yaxis\":{\"gridcolor\":\"white\",\"linecolor\":\"white\",\"ticks\":\"\",\"title\":{\"standoff\":15},\"zerolinecolor\":\"white\",\"automargin\":true,\"zerolinewidth\":2},\"scene\":{\"xaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2},\"yaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2},\"zaxis\":{\"backgroundcolor\":\"#E5ECF6\",\"gridcolor\":\"white\",\"linecolor\":\"white\",\"showbackground\":true,\"ticks\":\"\",\"zerolinecolor\":\"white\",\"gridwidth\":2}},\"shapedefaults\":{\"line\":{\"color\":\"#2a3f5f\"}},\"annotationdefaults\":{\"arrowcolor\":\"#2a3f5f\",\"arrowhead\":0,\"arrowwidth\":1},\"geo\":{\"bgcolor\":\"white\",\"landcolor\":\"#E5ECF6\",\"subunitcolor\":\"white\",\"showland\":true,\"showlakes\":true,\"lakecolor\":\"white\"},\"title\":{\"x\":0.05},\"mapbox\":{\"style\":\"light\"}}},\"title\":{\"text\":\"STAR Alignment Summary\"},\"xaxis\":{\"title\":{\"text\":\"Read Categories\"}},\"yaxis\":{\"title\":{\"text\":\"Percentage (%)\"}}},                        {\"responsive\": true}                    ).then(function(){\n",
              "                            \n",
              "var gd = document.getElementById('f03e518e-066e-427a-88ea-e071c7f49d44');\n",
              "var x = new MutationObserver(function (mutations, observer) {{\n",
              "        var display = window.getComputedStyle(gd).display;\n",
              "        if (!display || display === 'none') {{\n",
              "            console.log([gd, 'removed!']);\n",
              "            Plotly.purge(gd);\n",
              "            observer.disconnect();\n",
              "        }}\n",
              "}});\n",
              "\n",
              "// Listen for the removal of the full notebook cells\n",
              "var notebookContainer = gd.closest('#notebook-container');\n",
              "if (notebookContainer) {{\n",
              "    x.observe(notebookContainer, {childList: true});\n",
              "}}\n",
              "\n",
              "// Listen for the clearing of the current output cell\n",
              "var outputEl = gd.closest('.output');\n",
              "if (outputEl) {{\n",
              "    x.observe(outputEl, {childList: true});\n",
              "}}\n",
              "\n",
              "                        })                };                            </script>        </div>\n",
              "</body>\n",
              "</html>"
            ]
          },
          "metadata": {}
        }
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "584f89e0"
      },
      "source": [
        "The Plotly bar chart visualizes the STAR alignment summary based on the output from the `SRR15509125_Log.final.out` file. Each bar represents a different category of reads after alignment:\n",
        "\n",
        "*   **Uniquely mapped reads %**: This bar shows the percentage of reads that aligned to only one unique location in the reference genome. A high percentage here indicates a successful and specific alignment.\n",
        "*   **% of reads mapped to multiple loci**: This bar represents the percentage of reads that aligned to more than one location in the reference genome. These reads are often discarded or handled separately in downstream analysis.\n",
        "*   **% of reads mapped to too many loci**: This indicates the percentage of reads that aligned to an excessive number of locations, often suggesting repetitive sequences or low-quality reads.\n",
        "*   **% of reads unmapped: too short**: This bar shows the percentage of reads that did not align because they were too short to be confidently mapped to the reference genome.\n",
        "*   **% of reads unmapped: other**: This represents the percentage of reads that failed to align for reasons other than being too short or having too many mismatches. This could include issues with read quality, adapter contamination, or reads originating from sequences not present in the reference genome.\n",
        "*   **% of chimeric reads**: This bar shows the percentage of reads that aligned as chimeric, meaning they mapped to two different locations in the genome. This can indicate structural variations or gene fusions.\n",
        "\n",
        "Analyzing these percentages helps assess the quality of the sequencing data and the efficiency of the alignment process. In this specific plot, you can observe the proportion of reads that successfully mapped uniquely, as well as the reasons for reads not mapping."
      ]
    },
    {
      "cell_type": "markdown",
      "metadata": {
        "id": "041555c5"
      },
      "source": [
        "Here is a summary of the steps we have completed in this notebook:\n",
        "\n",
        "1.  **Setup and Installation**: We started by installing the necessary tools and libraries, including `build-essential`, `sra-toolkit`, `unzip`, and the STAR aligner. We downloaded and compiled STAR from its source.\n",
        "2.  **Data Download**: We downloaded the FASTQ files for the SRA accession SRR15509125 using `prefetch` and extracted the paired-end reads using `fasterq-dump`.\n",
        "3.  **Reference Genome Preparation**: We downloaded the reference genome sequence (chromosome 22) and the corresponding GTF annotation file from Ensembl. These files were then unzipped.\n",
        "4.  **STAR Genome Index Generation**: We used STAR to generate a genome index based on the downloaded reference genome and GTF file. This index is required for efficient alignment.\n",
        "5.  **STAR Alignment**: We performed the alignment of the FASTQ reads to the generated genome index using STAR. The output included a sorted BAM file containing the alignments and a gene counts file.\n",
        "6.  **Output Inspection**: We inspected the generated output files, including the sorted BAM file, the gene counts file, and the STAR log file, to check for successful execution and initial alignment statistics.\n",
        "7.  **Alignment Summary Visualization**: We parsed the STAR log file and created a Plotly bar chart to visualize the alignment summary statistics, such as the percentage of uniquely mapped reads, multi-mapped reads, and unmapped reads."
      ]
    }
  ]
}